client/landing/gutenboarding/onboarding-block/style-preview/preview.tsx

```diff
@@ -11,13 +12,110 @@ import { STORE_KEY } from '../../stores/onboard';
+	const [ previewHtml, setPreviewHtml ] = React.useState< string >();
+	const [ requestedFonts, setRequestedFonts ] = React.useState< Set< string > >( new Set() );
+
+	const { selectedDesign, siteVertical } = useSelect( select =>
+		select( STORE_KEY ).getState()
+	) as { selectedDesign: Design; siteVertical: SiteVertical };
+
+	const iframe = React.useRef< HTMLIFrameElement >( null );
+
+	React.useEffect(
+		() => {
+			const eff = async () => {
+				const [ { fontFamily: fontHeadings }, { fontFamily: fontBase } ] = fonts;
+				const templateUrl = `https://public-api.wordpress.com/rest/v1/template/demo/${ encodeURIComponent(
+					selectedDesign.slug
+				) }/${ encodeURIComponent( selectedDesign.slug ) }/`;
+				const url = addQueryArgs( templateUrl, {
+					language: 'en',
+					vertical: siteVertical.label,
+					font_headings: fontHeadings,
+					font_base: fontBase,
+				} );
+				let resp;
+				try {
+					resp = await window.fetch( url );
+					if ( resp.status < 200 || resp.status >= 300 ) {
+						throw resp;
+					}
+				} catch ( err ) {
+					if ( process.env.NODE_ENV !== 'production' ) {
+						// Disable reason: Log errors in development
+						// eslint-disable-next-line no-console
+						console.error( err );
+					}
+					setPreviewHtml( '<h1>Error loading preview.</h1>' );
+					setRequestedFonts( new Set() );
+					return;
+				}
+				const html = await resp.text();
+				setPreviewHtml( html );
+				setRequestedFonts( new Set( [ fontHeadings, fontBase ] ) );
+			};
+			eff();
+		},
+		// Disable reason: We'll handle font change elsewhere.
+		[ selectedDesign, siteVertical ] // eslint-disable-line react-hooks/exhaustive-deps
+	);
+
+	React.useEffect( () => {
+		if ( previewHtml ) {
+			const iframeDocument = iframe.current?.contentWindow?.document;
+			if ( iframeDocument ) {
+				iframeDocument.open();
+				iframeDocument.write( previewHtml );
+				iframeDocument.close();
+			}
+		}
+	}, [ previewHtml ] );
+
+	React.useEffect( () => {
+		const iframeWindow = iframe.current?.contentWindow;
+		if ( iframeWindow?.document.body ) {
+			const iframeDocument = iframeWindow.document;
+			const [ { fontFamily: headings }, { fontFamily: base } ] = fonts;
+
+			const baseURL = 'https://fonts.googleapis.com/css2';
+
+			// matrix: regular,bold * regular,italic
+			const axis = 'ital,wght@0,400;0,700;1,400;1,700';
+			const query = [];
+
+			// To replace state
+			const nextFonts = new Set( requestedFonts );
+
+			if ( ! requestedFonts.has( headings ) ) {
+				query.push( `family=${ encodeURI( headings ) }:${ axis }` );
+				nextFonts.add( headings );
+			}
+			if ( ! requestedFonts.has( base ) ) {
+				query.push( `family=${ encodeURI( base ) }:${ axis }` );
+				nextFonts.add( base );
+			}
+
+			if ( query.length ) {
+				const l = iframeDocument.createElement( 'link' );
+				l.rel = 'stylesheet';
+				l.type = 'text/css';
+				l.href = `${ baseURL }?${ query.join( '&' ) }&display=swap`;
+				iframeDocument.head.appendChild( l );
```

Comment:
Noting here that it'd be nice to encapsulate Google Fonts API handling. There seem to be a few libraries out there to do that ([#](https://www.npmjs.com/package/get-google-fonts#Example-Result)), but I haven't found one that's used widely enough to make sense to use.

There's one [Webpack plugin](https://www.npmjs.com/package/google-fonts-plugin) that seems relatively popular, but I'm not sure we want to add more chunks to the mix.

Reply:
These bits might/will change once core Global styles is ready; now it's just stying to stay in sync with what happens in wpcom-Global styles and it's ok to be more manual/disconnected.

There might be some core functions available later on for all this.